builtin/providers/kubernetes/resource_kubernetes_pod.go

```diff
@@ -0,0 +1,172 @@
+				Description: "Spec of the pod owned by the cluster",
+				Required:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: podSpecFields(),
+				},
+			},
+		},
+	}
+}
+func resourceKubernetesPodCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	metadata := expandMetadata(d.Get("metadata").([]interface{}))
+	spec, err := expandPodSpec(d.Get("spec").([]interface{}))
+	if err != nil {
+		return err
+	}
+
+	pod := api.Pod{
+		ObjectMeta: metadata,
+		Spec:       spec,
+	}
+
+	log.Printf("[INFO] Creating new pod: %#v", pod)
+	out, err := conn.CoreV1().Pods(metadata.Namespace).Create(&pod)
+
+	if err != nil {
+		return err
+	}
+	log.Printf("[INFO] Submitted new pod: %#v", out)
+	stateConf := &resource.StateChangeConf{
+		Target:  []string{"Running"},
+		Pending: []string{"Pending"},
+		Timeout: 5 * time.Minute,
+		Refresh: func() (interface{}, string, error) {
+			out, err := conn.CoreV1().Pods(metadata.Namespace).Get(metadata.Name)
+			if err != nil {
+				log.Printf("[ERROR] Received error: %#v", err)
+				return out, "Error", err
+			}
+
+			statusPhase := fmt.Sprintf("%v", out.Status.Phase)
+			log.Printf("[DEBUG] Pods %s status received: %#v", out.Name, statusPhase)
+			return out, statusPhase, nil
+		},
+	}
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return err
+	}
+	log.Printf("[INFO] Pod %s created", out.Name)
+
+	d.SetId(buildId(out.ObjectMeta))
+	return resourceKubernetesPodRead(d, meta)
+}
+
+func resourceKubernetesPodUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+	namespace, name := idParts(d.Id())
+	ops := patchMetadata("metadata.0.", "/metadata/", d)
+	if d.HasChange("spec") {
+		specOps, err := patchPodSpec("/spec", "spec", d)
+		if err != nil {
+			return err
+		}
+		ops = append(ops, specOps...)
+	}
+	data, err := ops.MarshalJSON()
+	if err != nil {
+		return fmt.Errorf("Failed to marshal update operations: %s", err)
+	}
+
+	log.Printf("[INFO] Updating  pod%s: %s", d.Id(), ops)
+	out, err := conn.CoreV1().Pods(namespace).Patch(name, pkgApi.JSONPatchType, data)
+	if err != nil {
+		return err
+	}
+	log.Printf("[INFO] Submitted updated pod: %#v", out)
+
+	d.SetId(buildId(out.ObjectMeta))
+	return resourceKubernetesPodRead(d, meta)
+}
+
+func resourceKubernetesPodRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+	namespace, name := idParts(d.Id())
+
+	log.Printf("[INFO] Reading pod %s", name)
+	pod, err := conn.CoreV1().Pods(namespace).Get(name)
+	if err != nil {
+		log.Printf("[DEBUG] Received error: %#v", err)
+		return err
+	}
+	log.Printf("[INFO] Received pod: %#v", pod)
+
+	err = d.Set("metadata", flattenMetadata(pod.ObjectMeta))
+	if err != nil {
+		return err
+	}
+
+	userSpec, err := expandPodSpec(d.Get("spec").([]interface{}))
+	if err != nil {
+		return err
+	}
+
+	err = d.Set("spec", flattenPodSpec(pod.Spec, userSpec))
```

Comment:
I'm not sure I understand this - why exactly do you need the config (`userSpec`) here? We should be able to filter out the internal volumes without it, I think - unless I'm missing something 🤔